javascript/ql/src/semmle/javascript/frameworks/PropertyProjection.qll

```diff
@@ -7,43 +7,23 @@
 
 import javascript
 
-/**
- * A property projection call such as `_.get(o, 'a.b')`, which is equivalent to `o.a.b`.
- */
-abstract class CustomPropertyProjection extends DataFlow::CallNode {
-  /**
-   * Gets the argument for the object to project properties from, such as `o` in `_.get(o, 'a.b')`.
-   */
-  abstract DataFlow::Node getObject();
-
-  /**
-   * Gets an argument that selects the properties to project, such as `'a.b'` in `_.get(o, 'a.b')`.
-   */
-  abstract DataFlow::Node getASelector();
-
-  /**
-   * Holds if this call returns the value of a single projected property, as opposed to an object that can contain multiple projected properties.
-   */
-  abstract predicate isSingletonProjection();
-}
-
 /**
  * A property projection call such as `_.get(o, 'a.b')`, which is equivalent to `o.a.b`.
  */
 class PropertyProjection extends DataFlow::CallNode {
```

Comment:
While we are at it, this class could also get the `Range` pattern description that you added to the `ClientRequest` above: "Extend this class to work with client request APIs for which there is already a model. ...."